src/MyConnection.h

```diff
@@ -6,6 +6,10 @@
 #include <boost/noncopyable.hpp>
 #include <boost/shared_ptr.hpp>
 
+const char* as_api_string(const Rcpp::Nullable<std::string>& cpp_string) {
```

Comment:
I thought this was the ideal situation for a macro, but hey I'm not a C/C++ programmer :cat: 

I can drop the commits myself if you want